app/javascript/components/network-floatingIPs-form/index.jsx

```diff
@@ -0,0 +1,114 @@
+import React, { useState, useEffect } from 'react';
+import MiqFormRenderer from '@@ddf';
+import PropTypes from 'prop-types';
+import createSchema from './network-floatingIPs-form.schema';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+import { networkProviders } from './helper';
+
+const NetworkFloatingIPsForm = ({
+  recordId,
+}) => {
+  const [{
+    ems, initialValues, isLoading, fields,
+  }, setState] = useState({
+    isLoading: !!recordId,
+    ems: [],
+  });
+
+  const submitLabel = !!recordId ? __('Save') : __('Add');
+
+  const emptySchema = (appendState = {}) =>
+    setState((state) => ({
+      ...state,
+      ...appendState,
+      fields: [],
+    }));
+
+  const loadSchema = (appendState = {}) => ({
+    data: {
+      // eslint-disable-next-line camelcase
+      form_schema: { fields },
+    },
+  }) =>
+    setState((state) => ({
+      ...state,
+      ...appendState,
+      fields,
+    }));
+
+  const onSubmit = (values) => {
+    miqSparkleOn();
+
+    const request = recordId ? API.patch(`/api/floating_ips/${recordId}`, values) : API.post('/api/floating_ips', values);
+    request.then(() => {
+      const message = sprintf(
+        recordId
+          ? __('New floating IPs "%s" has been successfully queued.')
+          : __('Add of floating IPs "%s" has been successfully queued.'),
+        values.fixed_ip_address,
+      );
+      miqRedirectBack(message, 'success', '/floating_ip/show_list');
+    }).catch(miqSparkleOff);
+  };
+
+  const onCancel = () => {
+    let url = '';
+    const message = sprintf(
+      recordId
+        ? __('Edit of floating IPs "%s" was canceled by the user.')
+        : __('Creation of new floating IPs was canceled by the user.'),
+      initialValues && initialValues.address,
+    );
+
+    if (recordId) {
+      url = `/floating_ip/show/${recordId}`;
+    } else {
+      url = '/floating_ip/show_list';
+    }
+
+    miqRedirectBack(message, 'warning', url);
+  };
+
+  useEffect(() => {
```

Comment:
@mengxiwuu Could you move useEffect call to top of the file after variable declaration for code consistency?